cmd/agent/subcommands/jmx/command.go

```diff
@@ -159,6 +160,11 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 			taggerimpl.Module(),
 			autodiscoveryimpl.Module(),
 			agent.Bundle(jmxloggerimpl.NewCliParams(cliParams.logFile)),
+			// InitSharedContainerProvider must be called before the application starts so the workloadmeta collector can be initiailized correctly.
+			// Since the tagger depends on the workloadmeta collector, we can not make the tagger a dependency of workloadmeta as it would create a circular dependency.
```

Comment:
Could we add a `TODO - component` here to track the cleanup work once we remove the cycle deps between workloadmeta and the tagger ?